keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -649,89 +629,182 @@ endpoints:
+namespaced name in the local cluster. There will be no change to existing
+behavior of the `cluster.local` zone.
+
+#### EndpointSlice
+
+When a `ServiceExport` is created, this will cause `EndpointSlice` objects for
+the underlying `Service` to be created in each importing cluster within the supercluster,
+associated with the derived `ServiceImport`. One or more `EndpointSlice`
+resources will exist for the exported `Service`, with each `EndpointSlice`
+containing only endpoints from a single source cluster. These `EndpointSlice`
+objects will be marked as managed by the supercluster service controller, so
+that the endpoint slice controller doesn’t delete them. `EndpointSlices` will
+have an owner reference to their associated `ServiceImport`.
+
+```
+<<[UNRESOLVED]>>
+We have not yet sorted out scalability impact here. We hope the upper bound for
+imported endpoints + in-cluster endpoints will be ~= the upper bound for
+in-cluster endpoints today, but this remains to be determined.
+<<[/UNRESOLVED]>>
+```
+
+#### Endpoint TTL
+
+To prevent stale endpoints from persisting in the event that a cluster becomes
+unreachable to the supercluster controller, each `EndpointSlice` is associated
+with a lease representing connectivity with its source cluster. The supercluster
+service controller is responsible for periodically renewing the lease so long as
+the connection with the source cluster is confirmed alive. A separate
+controller, that may run inside each cluster, is responsible for watching each
+lease and removing all remaining `EndpointSlices` associated with a cluster when
+that cluster’s lease expires.
 
 ## Constraints and Conflict Resolution
 
 Exported services are derived from the properties of each component service and
 their respective endpoints. However, some properties combine across exports
-better than others. They generally fall into two categories: global properties,
-and component-level properties.
-
+better than others.
 
 ### Global Properties
 
 These properties describe how the service should be consumed as a whole. They
 directly impact service consumption and must be consistent across all child
 services. If these properties are out of sync for a subset of exported services,
-there is no clear way to determine how a service should be accessed. **If any
-global properties have conflicts that can not be resolved, a condition will be
-set on the `ServiceExport` with a description of the conflict. The service will
-not be synced, and an error will be set on the status of each affected
-`ServiceExport` and any previously-derived `ServiceImports` will be deleted
-from each cluster in the supercluster.**
+there is no clear way to determine how a service should be accessed.
+
+Conflict resolution policy: **If any properties have conflicting values that can
+not simply be merged, a `ServiceExportConflict` condition will be set on the
+`ServiceExport` with a description of the conflict. The conflict will be
```

Comment:
"the ServicExport" - which one or all of them?

Reply:
clarified all